VideoLocker/src/main/java/org/edx/mobile/module/analytics/ISegmentImpl.java

```diff
@@ -540,21 +540,39 @@ public Properties trackUserLogout() {
     }
 
     /**
-     * This function is used send Screen View Details
-     * @param screenName
-     * @return 
+     * This function is used to send Screen View Details
+     *
+     * @param screenName The screen name to track
+     * @return
      */
     @Override
-    public Properties screenViewsTracking(String screenName) {
-        try{
-            SegmentAnalyticsEvent aEvent = new SegmentAnalyticsEvent();
-            aEvent.setAppNameContext();
-            tracker.screen("", screenName, aEvent.properties);
-            return aEvent.properties;
-        }catch(Exception e){
-            logger.error(e);
-            return null;
+    public Properties trackScreenView(String screenName) {
+        return trackScreenView(screenName, null, null);
+    }
+
+    /**
+     * This function is used to send Screen View Details with an extra event of screen view with
+     * more details
```

Comment:
This can be changed to "This function is used to send the screen tracking event, with an extra event for sending course id."